lib/Doctrine/Annotations/Metadata/AnnotationTarget.php

```
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Metadata;
+
+use Doctrine\Annotations\Annotation\Target;
+use function array_filter;
+use function assert;
+use function implode;
+
+/**
+ * @internal
+ */
+final class AnnotationTarget
+{
+    public const TARGET_CLASS      = 1;
+    public const TARGET_METHOD     = 2;
+    public const TARGET_PROPERTY   = 4;
+    public const TARGET_ANNOTATION = 8;
+    public const TARGET_ALL        = self::TARGET_CLASS
+        | self::TARGET_METHOD
+        | self::TARGET_PROPERTY
+        | self::TARGET_ANNOTATION;
+
+    private const LABELS = [
+        self::TARGET_CLASS      => 'CLASS',
+        self::TARGET_METHOD     => 'METHOD',
+        self::TARGET_PROPERTY   => 'PROPERTY',
+        self::TARGET_ANNOTATION => 'ANNOTATION',
+        self::TARGET_ALL        => 'ALL',
+    ];
+
+    /** @var int */
+    private $target;
+
+    public function __construct(int $target)
+    {
+        assert($target >= 0 && $target <= self::TARGET_ALL, 'Target out of bounds.');
```

Same as above, don't use an assertion but rather throw an exception.